sdk/core/Azure.Core.Experimental/README.md

```diff
@@ -1,3 +1,59 @@
 # Azure Core Experimental shared client library for .NET
 
 Azure.Core.Experimental contains types that are being evaluated and might eventually become part of Azure.Core, this library would always stay in a preview version and might allow breaking changes.
+
+## Binary Data
+### Overview
+ The `BinaryData` type provides a lightweight abstraction for a payload of bytes. This type integrates with [ObjectSerializer](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Serialization/ObjectSerializer.cs) to allow for serializing and deserializing payloads. It provides convenient helper methods to get out commonly used primitives, such as streams, strings, or bytes. The assumption when converting to and from string is that the encoding is UTF-8.
+ 
+ ### Data ownership
+ The ownership model of the underlying bytes varies depending on how the instance is constructed:
+ If created using the static factory method, `FromMemory(ReadOnlyMemory<byte>)`, the passed in bytes will be wrapped, rather than copied. This is useful in scenarios where performance is critical and/or ownership of the bytes is controlled completely by the consumer, thereby allowing the enforcement of whatever ownership model is needed.
+ 
+ If created using the `BinaryData(ReadOnlySpan<byte>)` constructor, `BinaryData` will maintain its own copy of the underlying bytes. This usage is geared more towards scenarios where the ownership of the bytes might be ambiguous to users of the consuming code. By making a copy of the bytes, the payload is guaranteed to be immutable. For all other constructors and static factory methods, BinaryData will assume ownership of the underlying bytes.
+
+ ### Usage
+ The main value of this type is its ability to easily convert from string to bytes to stream. This can greatly simplify API surface areas by exposing this type as opposed to numerous overloads or properties.
+ 
+ ```C# Snippet:BinaryDataHelloWorld
+var data = new BinaryData("some data");
+
+// there is an implicit cast defined for ReadOnlyMemory<byte>
+ReadOnlyMemory<byte> bytes = data;
+
+// there is also a Bytes property that holds the data
+bytes = data.Bytes;
+
+// ToString will decode the bytes using UTF-8
+Console.WriteLine(data.ToString()); // prints "some data"
+
+// you can also create BinaryData from a stream
+data = BinaryData.FromStream(new MemoryStream());
+
+// and convert it to a stream
+var stream = data.ToStream();
+```
+ 
+ It is also used to integrate with `ObjectSerializer`. Here is an [example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Message/MessageLiveTests.cs#L149) from Service Bus where `BinaryData` is used to represent a message body:
+ 
+```c#
```

Comment:
We'll have to figure something out when merging to core.

Reply:
Removed it and replaced with custom model defined in Azure.Core.Experimental test.